pallets/order-book/src/lib.rs

```diff
@@ -677,12 +679,27 @@ pub mod pallet {
             );
 
             let mut data = CacheDataLayer::<T>::new();
-            order_book.place_limit_order(order, &mut data)?;
+
+            let executed_orders_count = order_book.place_limit_order(order, &mut data)?;
 
             data.commit();
             <OrderBooks<T>>::insert(order_book_id, order_book);
 
-            Ok(().into())
+            // Note: be careful with changing the weight. The fee depends on it,
+            // the market-maker fee is charged for some weight, and the regular fee for none weight
+            let actual_weight = if executed_orders_count == 0 {
+                // if the extrinsic just places the limit order, the weight of the placing is returned
+                Some(<T as Config>::WeightInfo::place_limit_order())
+            } else {
+                // if the limit order was converted into market order, then None weight is returned
+                // this weight will be replaced with worst case weight - exchange_weight()
```

Comment:
Is this logic planned to be implemented in other PR? I don't see it here

Reply:
Also didn't we want to charge fee proportional to the executed number instead?

Reply:
Not propotional, just worst case.
It is already implemented, look at `#[pallet::weight(Pallet::<T>::exchange_weight())]`. This weight will be used if extrinsic returns `None` for the weight.

Reply:
i though about another piece of logic (fee calc); it was clarified to me that it's going to be resolved in a following PR